src/vs/workbench/contrib/search/browser/searchView.ts

```diff
@@ -1678,14 +1700,23 @@ export class SearchView extends ViewletPane {
 	}
 
 	private onFilesChanged(e: FileChangesEvent): void {
-		if (!this.viewModel || !e.gotDeleted()) {
+		if (this.searchConfig.sortOrder !== 'modified' && (!this.viewModel || !e.gotDeleted())) {
```

Comment:
Should this be `||`? Otherwise `const matches = this.viewModel.searchResult.matches();` below will throw when the search order is modified but the `viewModel` is `undefined`